src/main/java/com/enderio/base/common/hangglider/PlayerMovementHandler.java

```diff
@@ -0,0 +1,76 @@
+package com.enderio.base.common.hangglider;
+
+import com.enderio.api.glider.GliderMovementInfo;
+import com.enderio.api.integration.IntegrationManager;
+import com.enderio.base.common.advancement.UseGliderTrigger;
+import com.enderio.base.common.lang.EIOLang;
+import net.minecraft.network.chat.Component;
+import net.minecraft.server.level.ServerPlayer;
+import net.minecraft.world.entity.player.Player;
+import net.minecraft.world.phys.Vec3;
+import net.minecraftforge.event.TickEvent;
+import net.minecraftforge.eventbus.api.SubscribeEvent;
+import net.minecraftforge.fml.common.Mod;
+
+import java.util.Optional;
+
+@Mod.EventBusSubscriber
+public class PlayerMovementHandler {
+
+    /**
+     * {@linkplain net.minecraft.world.entity.LivingEntity#travel} 0.91 multiplicator
+     */
+    private static final double AIR_FRICTION_COEFFICIENT = 1/0.91D;
+
+    private static final double MOVEMENT_CHANGE_EFFECT = 0.05d;
+
+    @SubscribeEvent
+    public static void onPlayerTick(TickEvent.PlayerTickEvent playerTickEvent) {
+        Player player = playerTickEvent.player;
+        if (playerTickEvent.phase == TickEvent.Phase.START) {
+            Optional<GliderMovementInfo> gliderMovementInfoOpt = calculateGliderMovementInfo(player, true);
+            if (gliderMovementInfoOpt.isEmpty())
```

Comment:
`calculateGliderMovementInfo()` triggers a client message even when glider is not equipped